library/system/src/lib/yast2/clients/view_anymsg.rb

```diff
@@ -0,0 +1,308 @@
+#
+# ***************************************************************************
+
+require "yast/core_ext"
+
+require "yast2/popup"
+
+Yast.import "UI"
+Yast.import "CommandLine"
+Yast.import "Directory"
+Yast.import "FileUtils"
+Yast.import "Label"
+Yast.import "Package"
+
+module Yast
+  # Reads a \n separated list of filenames from
+  # /var/lib/YaST2/filenames
+  # Lines starting with "#" are ignored (comments)
+  # A line starting with "*" is taken as the default filename, the "*" is stripped
+  #
+  # All files are listed in an editable combo box, where the user can
+  # easily switch between files and even add a new file
+  #
+  # At finish, the list of filenames is written back to
+  # /var/lib/YaST2/filenames
+  # adapting the default line (starting with "*") accordingly.
+  #
+  # The default is either given as WFM::Args(0) or is the file last viewed.
+  class ViewAnymsgClient < Client
+    using Yast::CoreExt::AnsiString
+
+    # [String] Default list of log files
+    DEFAULT_FILENAMES = [
+      "/var/log/boot.log",
+      "/var/log/messages",
+      "/var/log/YaST2/y2log"
+    ].freeze
+
+    def main
+      textdomain "base"
+
+      # the command line description map
+      return CommandLine.Run("id" => "view_anymsg") if WFM.Args.first == "help"
+
+      # set up dialogue
+      UI.OpenDialog(
+        Opt(:decorated, :defaultsize),
+        dialog_content
+      )
+
+      # wait until user clicks "OK"
+      # check if ComboBox selected and change view accordingly
+      res = nil
+
+      loop do
+        # Fill the LogView with file content
+        UI.ChangeWidget(Id(:log), :Value, file_content(selected_filename))
+
+        heading = Builtins.sformat(_("System Log (%1)"), selected_filename)
+        UI.ChangeWidget(Id(:log), :Label, heading)
+
+        if start_journal?
+          res = :journal
+          break
+        end
+
+        # wait for user input
+        res = UI.UserInput
+
+        case res
+        when :ok, :cancel then break
+        when :custom_file
+          # adapt to combo box settings
+          new_file = UI.QueryWidget(Id(:custom_file), :Value)
+          self.selected_filename = new_file if !new_file.nil?
+        else
+          Builtins.y2milestone("bad UserInput (%1)", res)
+        end
+      end
+
+      write_new_filenames if res == :ok
+      UI.CloseDialog
+
+      Yast::WFM.CallFunction("journal") if res == :journal
+
+      true
+    end
+
+  private
+
+    def start_journal?
+      return false unless [nil, 0, -1].include?(FileUtils.GetSize(selected_filename))
+
+      res = Yast2::Popup.show(
+        _(
+          "Selected log file does not exist or is empty.\n" \
+          "Many system components now log into systemd journal.\n" \
+          "Do you want to start YaST module for systemd journal?"
+        ),
+        buttons: :yes_no,
+        focus:   :no
+      ) == :yes
+
+      return false unless res
+
+      if !Package.Installed("yast2-journal")
```

Comment:
for whole this part I expect that there have to be some existing method that do all that asking and checking availability, but do not find it. Any pointers?